### ISSUE TITLE
bug: zero attention-access threshold makes LRU treat negligible attention as a fresh access

### ISSUE THREAD

**Issue by umran666** [umran666/UniqKache#64]
### Bug and location

The policy contract at [src/uniqkache/policies/lru.py:15-28](https://github.com/umran666/UniqKache/blob/0b878a1acc3f2a84ea0ff16194456b8ed10191e8/src/uniqkache/policies/lru.py#L15-L28) defines access using non-negligible attention and explicitly explains why a positive threshold matters. However, [src/uniqkache/cache/metadata.py:139-167](https://github.com/umran666/UniqKache/blob/0b878a1acc3f2a84ea0ff16194456b8ed10191e8/src/uniqkache/cache/metadata.py#L139-L167) defaults to `threshold=0.0`, and [src/uniqkache/cache/kv_cache.py:433-455](https://github.com/umran666/UniqKache/blob/0b878a1acc3f2a84ea0ff16194456b8ed10191e8/src/uniqkache/cache/kv_cache.py#L433-L455) neither exposes nor passes a positive threshold.

Ordinary softmax produces positive probabilities even for effectively unattended keys. Those keys are consequently marked accessed on every decode step.

### Reproduction

```python
import torch
from uniqkache import CacheConfig, KVCache, build_policy

cache = KVCache(CacheConfig(1, 1, 4, dtype=torch.float32, capacity=4),
                policy=build_policy("lru"))
kv = torch.ones(1, 1, 4, 4)
cache.append(0, kv, kv)
cache.advance()
cache.note_attention(0, torch.tensor([[[[1.0, 1e-30, 1e-30, 1e-30]]]]))
print(cache.store.layer(0).metadata.last_access.tolist())
cache.advance()
cache.append(0, kv[:, :, :1], kv[:, :, :1])
print(cache.store.layer(0).metadata.positions.tolist())
```

Observed: all access timestamps become `[1,1,1,1]`; the next append evicts position 0 and retains `[1,2,3,4]`, despite position 0 receiving essentially all attention.

### Impact and expected behavior

With positive softmax probabilities, LRU reduces to positional tie-breaking instead of tracking meaningful access recency. This undermines its use as a distinct policy baseline and also contaminates recency features consumed by other policies.

Wire a meaningful, reproducibly recorded access threshold into the attention-recording path and test negligible versus material attention. This is independent of the float32 ranking precision problem tracked in #42.

### Verification

Reviewed commit: `0b878a1acc3f2a84ea0ff16194456b8ed10191e8`. Reproduced on Windows with Python 3.10 and PyTorch 2.6.0+cu124; CPU unless otherwise specified. Synthetic-model results below are correctness diagnostics, not language-model quality claims.
